middleware.ts

```diff
@@ -161,9 +134,47 @@ const isAuthenticatedPath = (pathname: string) =>
     authenticatedPath.test({ pathname }),
   );
 
-export const middleware = async (req: NextRequest) => {
+const httpLoggers = {
+  req: (req: NextRequest) => {
+    const correlationId = v4();
+    req.headers.set(HEADERS.CORRELATION_ID, correlationId);
+    logger.http('Incoming request', {
+      ...logger.utils.formatRequest(req),
+      correlationId,
+    });
+  },
+  res: (req: NextRequest, res: NextResponse) =>
+    logger.http(
+      'Outgoing response - PLEASE NOTE: this represents a snapshot of the response as it exits the middleware, changes made by other server code (eg getServerSideProps) will not be shown',
+      {
+        ...logger.utils.formatResponse(res),
+        correlationId: req.headers.get(HEADERS.CORRELATION_ID),
+      },
+    ),
+};
+
+type LoggerType = keyof typeof httpLoggers;
+
+const urlsToSkip = ['/_next/', '/assets/', '/javascript/'];
+
+const getConditionalLogger = (req, type: LoggerType) => {
   const userAgentHeader = req.headers.get('user-agent') || '';
-  if (!userAgentHeader.startsWith('ELB-HealthChecker')) logRequest(req);
-  const { pathname } = req.nextUrl;
-  if (isAuthenticatedPath(pathname)) return authenticateRequest(req);
+  return userAgentHeader.startsWith('ELB-HealthChecker') ||
+    urlsToSkip.some((url) => req.nextUrl.pathname.startsWith(url))
+    ? () => undefined
+    : httpLoggers[type];
+};
```

Comment:
I think this could be a bit easier to read with an if statement and multiple returns. But otherwise lgtm - approved

Reply:
That is fair, I've got another PR to follow this so I've made this change there.